apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/interceptor/AccessTokenProvider.kt

```diff
@@ -0,0 +1,6 @@
+package com.apollographql.apollo.interceptor
+
+interface AccessTokenProvider {
+  suspend fun currentAccessToken(): String
+  suspend fun newAccessToken(previousToken: String): String
```

Comment:
nit: should we call `currentToken()` and `renewToken(previousToken: String): String` accordingly ?

Reply:
nit: maybe we should add typealias:
```
typealias AccessToken = String
```

Reply:
Also I was thinking we need to provide description that it must be coroutine / thread safe. As there could be a race condition when several threads will call `newAccessToken `

Reply:
Or we should provide this thread safety via mutex
```
mutex.withLock {
        val token = accessTokenProvider.provide() 
     }
```

Reply:
Yes to all of that 💯